File: app/src/main/java/com/khmelenko/lab/varis/log/TextLeaf.java
package com.khmelenko.lab.varis.log;

final class TextLeaf implements LogEntryComponent {

    private FormattingOptions mOptions;
    private String mText = "";

    TextLeaf() {
        this("");
    }

    TextLeaf(String text) {
        this(text, FormattingOptions.fromAnsiCodes());
    }

    TextLeaf(FormattingOptions options) {
        this("", options);
    }

    public TextLeaf(String text, FormattingOptions options) {
        mText = text;
        mOptions = options;
    }

    public FormattingOptions getOptions() {
        return mOptions;
    }

    public void setOptions(FormattingOptions options) {
        mOptions = options;
    }

    public String getText() {
        return mText;
    }

    public void setText(String text) {
        mText = text;
    }

    @Override
    public String toHtml() {
        String result = mText.replaceAll("\\r\\n?|\\n", "<br />");
        if (mOptions.getTextColor() != null) {
            result = "<font color='" + mOptions.getTextColor() + "'>" + result + "</font>";
        } else {
            result = "<font color='#FFF'>" + result + "</font>";
        }
        if (mOptions.isBold()) {
            result = "<b>" + result + "</b>";
        }
        if (mOptions.isItalic()) {
            result = "<i>" + result + "</i>";
        }
        return result;
    }

    @Override
    public String toString() {
        return "TextLeaf{" +
                "options=" + mOptions +
                ", text='" + mText + '\'' +
                '}';
    }
}

File: app-v3/src/main/java/com/khmelenko/lab/varis/log/TextLeaf.kt
package com.khmelenko.lab.varis.log

internal class TextLeaf @JvmOverloads constructor(
        var text: String = "",
        var options: FormattingOptions = FormattingOptions.fromAnsiCodes()
) : LogEntryComponent {

    constructor(options: FormattingOptions) : this("", options)

    override fun toHtml(): String {
        var result = text.replace("\\r\\n?|\\n".toRegex(), "<br />")
        if (options.textColor != null) {
            result = "<font color='" + options.textColor + "'>" + result + "</font>"
        } else {
            result = "<font color='#FFF'>$result</font>"
        }
        if (options.isBold) {
            result = "<b>$result</b>"
        }
        if (options.isItalic) {
            result = "<i>$result</i>"
        }
        return result
    }

    override fun toString(): String {
        return "TextLeaf{" +
                "options=" + options +
                ", text='" + text + '\''.toString() +
                '}'.toString()
    }
}
